test/unit/project_type/test_git.py

```diff
@@ -79,20 +79,41 @@ def test_execute_command_in_project_type_specifies_cwd_if_doesnt_exist(self):
             start_new_session=ANY,
         )
 
-    def test_get_full_repo_directory(self):
+    @genty_dataset(
+        regular_path=(
+            'http://scm.example.com/path/to/project', 
+             join(
+                'scm.example.com',
+                'path',
+                'to',
+                'project',
+            )
+        ),
+        with_netloc=(
+            'ssh://scm.dev.box.net:12345/awesome-project',
+            join(
+                'scm.dev.box.net12345',
+                'awesomeproject',
+            )
+        ),
+        no_netloc=(
+            'git.dev.box.net:Productivity/ClusterRunnerHealthCheck',
+            join(
+                'git.dev.box.net',
+                'Productivity',
+                'ClusterRunnerHealthCheck',
+            )
+        ),
+    )
+    def test_get_full_repo_directory(self, url, expected_repo_path_without_base):
         Configuration['repo_directory'] = join(expanduser('~'), '.clusterrunner', 'repos')
-        url = 'http://scm.example.com/path/to/project'
-
-        actual_repo_sys_path = Git.get_full_repo_directory(url)
-
-        expected_repo_sys_path = join(
-            Configuration['repo_directory'],
-            'scm.example.com',
-            'path',
-            'to',
-            'project',
+        expected_repo_path = join(
+            Configuration['repo_directory'], 
+            expected_repo_path_without_base
```

Comment:
It's good to leave the trailing comma on the last element when you split the args over multiple lines.